Composer/packages/lib/shared/src/lgUtils/parsers/parseLgTemplateRef.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+import { LgTemplateRefString } from '../models/stringTypes';
+import LgTemplateRef from '../models/LgTemplateRef';
+
+import { LgTemplateRefPattern } from './lgPatterns';
+import parseLgParamString from './parseLgParamString';
+
+export default function parseLgTemplateRef(inputString: LgTemplateRefString): LgTemplateRef | null {
+  if (typeof inputString !== 'string') return null;
```

Comment:
if this based on extractTemplateRefs 
{
var refs = extractTempalteRefs(inputString)

if (res.length != 1) return null;
if (res[0].toString().length != input.length) return null;

return res[0];
}

Reply:
there is a trick here:
```ts
const input = '[greeting]'
res[0].toString() === '[greeting()]' !== '[greeting]'
```
So I didn't invoke extract inside parse, instead, I just put them together to provide the flexibility that removing either of them.